functions-and-operators: Improve format, add vitess_hash (#6558)
pingcap/docs: functions-and-operators/tidb-functions.md

# TiDB Specific Functions

The following functions are TiDB extensions, and are not present in MySQL:

> | Function name | Feature description |

Reviewer: ```suggestion
| Function name | Function description |
```

| :-------------- | :------------------------------------- |

> | `TIDB_BOUNDED_STALENESS()` | The `TIDB_BOUNDED_STALENESS` function is an internal function of TiDB. |

Reviewer: I don't think this sentence is very meaningful. See the function description in https://github.com/pingcap/docs/blob/master/as-of-timestamp.md#syntax : If you want to specify a time range, you can use the TIDB_BOUNDED_STALENESS() function in the clause. When this function is used, TiDB selects a suitable timestamp within the specified time range.

Author: I copied it from the previous text.

I can change it to "TIDB_BOUNDED_STALENESS() instructs TiDB to read the data as new as possible within the time range. See also: [Read Historical Data Using the `AS OF TIMESTAMP` Clause](/as-of-timestamp.md)"

Reviewer: LGTM

Author: OK changed.

| [`TIDB_DECODE_PLAN(str)`](#tidb_decode_plan) | The `TIDB_DECODE_PLAN` function can be used to decode a TiDB execution plan. |
| `TIDB_IS_DDL_OWNER()` | The `TIDB_IS_DDL_OWNER` function can be used to check whether or not the TiDB instance you are connected to is the one that is the DDL Owner. The DDL Owner is the TiDB instance that is tasked with executing DDL statements on behalf of all other nodes in the cluster. |
| [`TIDB_PARSE_TSO(num)`](#tidb_parse_tso) | The `TIDB_PARSE_TSO` function can be used to extract the physical timestamp from a TiDB TSO timestamp. See also: [`tidb_current_ts`](/system-variables.md#tidb_current_ts). |
| [`TIDB_VERSION()`](#tidb_version) | The `TIDB_VERSION()` function returns the TiDB version with additional build information. |
| [`TIDB_DECODE_SQL_DIGESTS(digests, stmtTruncateLength)`](#tidb_decode_sql_digests) | The `TIDB_DECODE_SQL_DIGESTS()` function is used to query the normalized SQL statements (a form without formats and arguments) corresponding to the set of SQL digests in the cluster. |

> | `VITESS_HASH(str)` | Returns the hash of a string that is compatible with Vitess' HASH function. This is intended to help in migrations from Vitess. |

Reviewer: ```suggestion
| `VITESS_HASH(str)` | The `VITESS_HASH(str)` function returns the hash of a string that is compatible with Vitess' `HASH` function. This is intended to help the data migration from Vitess. |
```

> ## Examples

Reviewer: ```suggestion
## Examples

This section provides examples for some of the functions above.
```

1 row in set (0.001 sec)
```

### TIDB_DECODE_PLAN

> Plans can be found in encoded form in the slow query log. The `TIDB_DECODE_PLAN()` function is then used to reverse the plan into a human-readable form.

Reviewer: ```suggestion
You can find TiDB execution plans in encoded form in the slow query log. The `TIDB_DECODE_PLAN()` function is then used to decode the encoded plans into a human-readable form.
```

> This is useful because the plan is captured at the time the statement executes. Re-executing the statement in `EXPLAIN` might produce different results as data distribution and statistics will evolve over time.

Reviewer: ```suggestion
This function is useful because a plan is captured at the time the statement is executed. Re-executing the statement in `EXPLAIN` might produce different results as data distribution and statistics evolves over time.
```

1 row in set (0.00 sec)
```

### MySQL compatibility

> The `TIDB_VERSION` function is TiDB-specific and not compatible with MySQL. If MySQL compatibility is required, you can also use `VERSION` to get version information, but the result does not contain build details.

Reviewer 2: I would prefer to keep this info somehere. 

Another compatible way of doing this is via the `version%` variables:
```
 MySQL  127.0.0.1:4000 ssl  test  SQL > show variables like 'version%';
+-------------------------+---------------------------------------------------------------------------+
| Variable_name           | Value                                                                     |
+-------------------------+---------------------------------------------------------------------------+
| version                 | 5.7.25-TiDB-v5.3.0-alpha-228-g74b329402                                   |
| version_comment         | TiDB Server (Apache License 2.0) Enterprise Edition, MySQL 5.7 compatible |
| version_compile_machine | x86_64                                                                    |
| version_compile_os      | osx10.8                                                                   |
+-------------------------+---------------------------------------------------------------------------+
4 rows in set, 1 warning (0.0029 sec)
```

Author: I included a briefer version of this text in the table:

> | [`TIDB_VERSION()`](#tidb_version) | The `TIDB_VERSION()` function returns the TiDB version with additional build information. |

i.e. The title of this page is TiDB specific syntax. The previous text was too verbose talking about "not being MySQL compatible" (that part should be obvious).

Reviewer 2: The thing I would keep in there is that people can use `VERSION()` instead of `TIDB_VERSION()` if they want compatibility. If you don't think that makes sense, feel free to let it out.
